dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java

```diff
@@ -139,27 +142,77 @@ protected ItemDAOImpl() {
+                return criteriaBuilder.equal(subqueryRoot.get(MetadataValue_.value), val);
+            }
+        },
+        not_equals {
+            public Predicate buildPredicate(String val, CriteriaBuilder criteriaBuilder,
+                                            Root<MetadataValue> subqueryRoot) {
+                return criteriaBuilder.notEqual(subqueryRoot.get(MetadataValue_.value), val);
+            }
+        },
+        like {
+            public Predicate buildPredicate(String val, CriteriaBuilder criteriaBuilder,
+                                            Root<MetadataValue> subqueryRoot) {
+                return criteriaBuilder.like(subqueryRoot.get(MetadataValue_.value), val);
+            }
+        },
+        not_like {
+            public Predicate buildPredicate(String val, CriteriaBuilder criteriaBuilder,
+                                            Root<MetadataValue> subqueryRoot) {
+                return criteriaBuilder.notLike(subqueryRoot.get(MetadataValue_.value), val);
+            }
+        },
+        contains {
+            public Predicate buildPredicate(String val, CriteriaBuilder criteriaBuilder,
+                                            Root<MetadataValue> subqueryRoot) {
+                return criteriaBuilder.like(subqueryRoot.get(MetadataValue_.value), "%" + val + "%");
+            }
+        },
+        doesnt_contain {
+            public Predicate buildPredicate(String val, CriteriaBuilder criteriaBuilder,
+                                            Root<MetadataValue> subqueryRoot) {
+                return criteriaBuilder.notLike(subqueryRoot.get(MetadataValue_.value), "%" + val + "%");
+            }
+        },
+        exists {
+            public Predicate buildPredicate(String val, CriteriaBuilder criteriaBuilder,
+                                            Root<MetadataValue> subqueryRoot) {
+                return OP.equals.buildPredicate(val, criteriaBuilder, subqueryRoot);
+            }
+        },
+        doesnt_exist {
+            public Predicate buildPredicate(String val, CriteriaBuilder criteriaBuilder,
+                                            Root<MetadataValue> subqueryRoot) {
+                return OP.not_equals.buildPredicate(val, criteriaBuilder, subqueryRoot);
+            }
+        };
+        public abstract Predicate buildPredicate(String val, CriteriaBuilder criteriaBuilder,
+                                                 Root<MetadataValue> subqueryRoot);
+    }
 
     @Override
     public Iterator<Item> findByMetadataQuery(Context context, List<List<MetadataField>> listFieldList,
                                               List<String> query_op, List<String> query_val, List<UUID> collectionUuids,
                                               String regexClause, int offset, int limit) throws SQLException {
-        Criteria criteria = createCriteria(context, Item.class, "item");
-        criteria.setFirstResult(offset);
```

Comment:
I don't see where offset and limit are applied, in the new code.

Reply:
The offset and limit are applied when calling the list method in the return line of this findByMetadataQuery method: https://github.com/DSpace/DSpace/pull/2111/files/316d9f08e577fe8cbee7b3ca850c1099959da6c8#diff-47280133d16cde27773db271f5386298R244
However thank you for asking about this because I noticed that I accidentally swapped the limit and offset around initially, this should be fixed in the newest commit.